packages/labs/test/selectTests.tsx

```diff
@@ -98,39 +98,127 @@ describe("<Select>", () => {
         assert.strictEqual(wrapper.state("query"), "");
     });
 
+    it("if query is non-empty, the input value will stay in sync with the query prop", () => {
+        const query = "nailed it";
+        const wrapper = select({ query });
+        const input = wrapper.find("input");
+
+        assert.equal(input.prop("value"), query);
+        assert.equal(wrapper.state("query"), query);
+
+        (input.getDOMNode() as HTMLInputElement).value = "some other value";
+        input.simulate("change");
```

Comment:
I'm not sure this works like you think it does. I thought you had to specify event data for anything to happen... 

`input.simulate("change", { currentTarget: { value: "..." } })` 

either way, I find it cleaner and less hacky to use the eventData approach.

Reply:
I tried `input.simulate("change", { currentTarget: { value: "..." } })` and `input.simulate("change", { target: { value: "..." } })` and even `input.simulate("change", { currentTarget: { value: "..." }, target: { value: "..." } })` with no luck